### DISCUSSION TITLE
Code for simplifications and experiments in Hyper-optimized tensor network contraction

### DISCUSSION THREAD

**Discussion by chrisstaudt** [jcmgray/cotengra#19] [General]
Hello,

thank you for publishing this library, it looks really interesting!

I just read your paper "Hyper-optimized tensor network contraction" and was looking for the code for the simplifications  experiments mentioned in that paper, but couldn't find them in this repository. While I am interested in both, I would be most interested in the simplifications. If you could share anything regarding this, that would be really great. It doesn't has to be in a nice shape, I know how easily research code can get messy. 

In any case thanks again!

**Comment by jcmgray** [comment]
Hi @CSNWEB, the tensor network simplification stuff is all in [`quimb`](https://quimb.readthedocs.io/en/latest/). Specifically:

- [`full_simplify`](https://quimb.readthedocs.io/en/latest/autoapi/quimb/tensor/index.html#quimb.tensor.TensorNetwork.full_simplify)
- [`compress_simplify`](https://quimb.readthedocs.io/en/latest/autoapi/quimb/tensor/index.html#quimb.tensor.TensorNetwork.compress_simplify) which is more experimental and unpublished..

There isn't any proper examples or documentation I'm afraid - the main thing to note for correctness of the results is that you might need to specify `output_inds=` if the TN has any hyper indices.
